tests/unittest_brain_pathlib.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+import astroid
+from astroid import bases
+from astroid.const import PY310_PLUS
+
+if not PY310_PLUS:
+    pytest.skip(
+        "The parents sequence supports slices since 3.10", allow_module_level=True
+    )
+
+
+def test_inference_parents_subscript_index():
+    """Test inference of ``pathlib.Path.parents``, accessed by index."""
+    name_node = astroid.extract_node(
+        """
+    from pathlib import Path
+
+    current_path = Path().resolve()
+    parent_path = current_path.parents[2]
+    parent_path
+    """
+    )
+    inferred = next(name_node.infer())
```

Comment:
Could you use `list` here and assert that `len(list) == 1`. We want to make sure that there is no ambiguity in the inference here and `next()` does not tell us that.

Reply:
Done! Used `.inferred()` instead of `.infer()` and checked the length of that, because that's what astroid/tests/unittest_brain_dataclasses.py does, although I'm not entirely sure what the difference is (`.inferred()` leverages an inference cache?).